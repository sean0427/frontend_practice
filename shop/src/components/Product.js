import React from 'react';

export default class Product extends React.PureComponent {
    static defaultProps = {
        imageSource: '',
        name: '',
        price: 0,
    };

    render() {
        const { imageSource, name, price } = this.props;

        return (
            <section className="product-item">
                <img src={ imageSource } alt="product image" />
                <h5>{ name }</h5>
                <em>{ price }</em>
            </section>
        );
    }
}
