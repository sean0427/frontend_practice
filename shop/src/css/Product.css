.products-list {
    display: flex;
    flex-wrap: wrap;
}

.product-item {
    display: flex;
    width: 200px;
    border: 2px solid;
    border-radius: 5px;
    border-color: #192838;
    margin: 1em;
    padding: 1em;
    background-color: #fffefc;
}
.product-item:hover {
    box-shadow: 1px 1px #5d6873;
}

.dollar::before {
    content: "$";
}
