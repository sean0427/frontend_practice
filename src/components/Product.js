import React from 'react';

export default class Product extends React.PureComponent {
    static defaultProps = {
        name: '',
        imageSource: '',
        price: 0,
        describe: '',
        product_id: 0,
        discounted_product_id: 0,
        discount: 1, 
    };

    render() {
        const { imageSource, name, price, discount, describe } = this.props;

        return (
            <section className="product-item">
                <img src={ imageSource } />
                <h5>{ name }</h5>
                <div className="describe">{ describe }</div>
                <em className="dollar">
                    { discount == 1 ? price : discount * price / 100 }
                </em>
            </section>
        );
    }
}
