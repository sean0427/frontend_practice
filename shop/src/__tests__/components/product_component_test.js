import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Product from '../../components/Product';

const TEST_PROPS = { 
    imageSource: "test.test/test.jpg",
    name: "PRODUCT NAME",
    price: 10000,
};

describe('Test Component Product', () => {
    it('render defualt', () => {
        const component = shallow(<Product />);

        expect(component).toHaveLength(1);
        expect(component.find('section')).toHaveLength(1);
        expect(component.find('img')).toHaveLength(1);
        expect(component.find('h5')).toHaveLength(1);
        expect(component.find('em')).toHaveLength(1);
    });

    it('render with props input', () => {
        const component = shallow(<Product />);

        component.setProps(TEST_PROPS);

        expect(component.find('img').prop('src')).toEqual(TEST_PROPS.imageSource);
        expect(component.find('h5').text()).toEqual(TEST_PROPS.name);
        expect(component.find('em').text()).toEqual(String(TEST_PROPS.price));

    });

    it('check change by snapshot', () => {
        const json = renderer.create(<Product />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
