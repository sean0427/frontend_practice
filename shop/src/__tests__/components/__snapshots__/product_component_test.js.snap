// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Product check change by snapshot 1`] = `
<section
  className="product-item"
>
  <img
    alt="product image"
    src=""
  />
  <h5>
    
  </h5>
  <em>
    0
  </em>
</section>
`;

exports[`Test Component Product snapshot 1`] = `
<section
  className="product-item"
>
  <img
    alt="product image"
    src=""
  />
  <h5>
    
  </h5>
  <em>
    0
  </em>
</section>
`;
